test/gtest/uct/uct_test.cc

```diff
@@ -32,7 +36,8 @@ resource::resource(uct_component_h component, const std::string& md_name,
                    const ucs_cpu_set_t& local_cpus, const std::string& tl_name,
                    const std::string& dev_name, uct_device_type_t dev_type) :
                    component(component), md_name(md_name), local_cpus(local_cpus),
-                   tl_name(tl_name), dev_name(dev_name), dev_type(dev_type)
+                   tl_name(tl_name), dev_name(dev_name), variant_name(""),
+                   dev_type(dev_type), variant(DEFAULT_VARIANT)
```

Comment:
looks like, we need to declare those ctors in `struct resource { }`
or i'm missing smth?

Reply:
i did not change constructors semantic, just initialized variants to default values

Reply:
do you think it is worse to define another constructor with variants?
I did not do it, because it is needed in just one place co far

Reply:
sorry, I was inattentive - I thought you add new parameters to ctor